packages/api/src/controllers/dashboard.ts

```diff
@@ -93,27 +110,54 @@ export async function updateDashboardAndAlerts(
       team: teamId,
     },
     {
-      ...dashboard,
-      tags: dashboard.tags && uniq(dashboard.tags),
+      ...updates,
+      tags: updates.tags && uniq(updates.tags),
     },
     { new: true },
   );
   if (updatedDashboard == null) {
     throw new Error('Could not update dashboard');
   }
 
-  // Delete related alerts
-  const deletedTileIds = differenceBy(
-    oldDashboard?.tiles || [],
-    updatedDashboard?.tiles || [],
-    'id',
-  ).map(c => c.id);
+  // Update related alerts
```

Comment:
style: I'd suggest to move all alerting mutation codes from here to another controller